src/librbd/crypto/CryptoObjectDispatch.cc

```diff
@@ -115,28 +113,27 @@ void CryptoObjectDispatch<I>::shut_down(Context* on_finish) {
 
 template <typename I>
 bool CryptoObjectDispatch<I>::read(
-    uint64_t object_no, const io::Extents &extents,
-    IOContext io_context, int op_flags, const ZTracer::Trace &parent_trace,
-    ceph::bufferlist* read_data, io::Extents* extent_map, uint64_t* version,
+    uint64_t object_no, io::ReadExtents* extents, IOContext io_context,
+    int op_flags, const ZTracer::Trace &parent_trace, uint64_t* version,
     int* object_dispatch_flags, io::DispatchResult* dispatch_result,
     Context** on_finish, Context* on_dispatched) {
   auto cct = m_image_ctx->cct;
   ldout(cct, 20) << data_object_name(m_image_ctx, object_no) << " "
-                 << extents << dendl;
+                 << *extents << dendl;
   ceph_assert(m_crypto != nullptr);
 
-  if (version != nullptr || extents.size() != 1) {
+  if (version != nullptr || extents->size() != 1) {
```

Comment:
Nit: finally, same comment here so that we support multiple extents throughout the stack